Frameworks/UIKit/UIView.mm

```diff
@@ -221,22 +281,25 @@ - (bool)_processGesturesForTouch:(UITouch*)touch event:(UIEvent*)event touchEven
         }
     }
 
-    //  Removed/reset failed/done gestures
-    for (int i = 0; i < viewDepth; i++) {
-        UIGestureRecognizer* curgesture = recognizers[i];
-        UIGestureRecognizerState state = (UIGestureRecognizerState)[curgesture state];
+    //  Removed/reset failed/done gestures, including gestures and DMGestures
+    for (int i = 0; i < 2; i++) {
+        int count = (i == 0) ? gestureCount : dmGestureCount;
```

Comment:
kind of strange; how about pulling this out into a shared function or lambda that gets called for both arrays?